virtcontainers/clh.go

```diff
@@ -0,0 +1,1260 @@
+		clh.Logger().Info("Invalid socket path for cloud-hypervisor")
+		return nil
+	}
+	clh.socketPath = socketPath
+
+	clh.Logger().WithField("function", "createSandbox").Info("creating Sandbox")
+
+	// No need to return an error from there since there might be nothing
+	// to fetch if this is the first time the hypervisor is created.
+	if err := clh.store.Load(store.Hypervisor, &clh.state); err != nil {
+		clh.Logger().WithField("function", "createSandbox").WithError(err).Info("No info could be fetched")
+	}
+
+	// Set initial memomory size of the cloud hypervisor
+	clh.cliBuilder.SetMemory(&CLIMemory{
+		memorySize:  clh.config.MemorySize,
+		backingFile: "/dev/shm",
+	})
+	// Set initial amount of cpu's for the cloud hypervisor
+	clh.cliBuilder.SetCpus(&CLICpus{
+		cpus: clh.config.NumVCPUs,
+	})
+
+	// Add the kernel path
+	kernelPath, err := clh.config.KernelAssetPath()
+	if err != nil {
+		return err
+	}
+	clh.cliBuilder.SetKernel(&CLIKernel{
+		path: kernelPath,
+	})
+
+	// First take the default parameters defined by this driver
+	clh.cliBuilder.AddKernelParameters(clhKernelParams)
+
+	// Followed by extra debug parameters if debug enabled in configuration file
+	if clh.config.Debug == true {
+		clh.cliBuilder.AddKernelParameters(clhDebugKernelParams)
+	}
+
+	// Followed by extra debug parameters defined in the configuration file
+	clh.cliBuilder.AddKernelParameters(clh.config.KernelParams)
+
+	// set random device generator to hypervisor
+	clh.cliBuilder.SetRng(&CLIRng{
+		src:   clh.config.EntropySource,
+		iommu: false,
+	})
+
+	// Add the hybrid vsock device to hypervisor
+	clh.cliBuilder.SetVsock(&CLIVsock{
+		cid:        3,
+		socketPath: clh.socketPath,
+		iommu:      false,
+	})
+
+	// set the initial root/boot disk of hypervisor
+	imagePath, err := clh.config.ImageAssetPath()
+	if err != nil {
+		return err
+	}
+
+	if imagePath != "" {
+		clh.cliBuilder.SetDisk(&CLIDisk{
+			path:  imagePath,
+			iommu: false,
+		})
+	}
+
+	// set the virtio-fs to the hypervisor
+	vfsdSockPath, err := clh.vfsdFSSocketPath(clh.id)
+	if err != nil {
+		return err
+	}
+	clh.cliBuilder.SetFs(&CLIFs{
+		tag:        "kataShared",
+		socketPath: vfsdSockPath,
+		queues:     1,
+		queueSize:  512,
+	})
+
+	// set the serial console to the cloud hypervisor
+	if clh.config.Debug {
+		serialPath, err := clh.serialPath(clh.id)
+		if err != nil {
+			return err
+		}
+		clh.cliBuilder.SetSerial(&CLISerialConsole{
+			consoleType: cctFILE,
+			filePath:    serialPath,
+		})
+	}
+	clh.cliBuilder.SetConsole(&CLIConsole{
+		consoleType: cctOFF,
+	})
+
+	return nil
+}
+
+func (clh *cloudHypervisor) startSandbox(timeout int) error {
+	span, _ := clh.trace("startSandbox")
+	defer span.Finish()
+
+	clh.Logger().WithField("function", "startSandbox").Info("starting Sandbox")
+
+	vmPath := filepath.Join(store.RunVMStoragePath(), clh.id)
+	err := os.MkdirAll(vmPath, store.DirMode)
+	if err != nil {
+		return err
+	}
+
+	if clh.config.SharedFS == config.VirtioFS {
```

Comment:
We should error out if `SharedFS` is not set to `VirtioFS` with Cloud Hypervisor.

Reply:
it's hard-wired in config.go now. but I can add one more check here.

Reply:
I see this is done, thanks.